controllers/operator/routes.go

```diff
@@ -277,13 +265,41 @@ func (r *AuthenticationReconciler) handleRoutes(ctx context.Context, instance *o
 	return
 }
 
+func (r *AuthenticationReconciler) removeIdauth(ctx context.Context, instance *operatorv1alpha1.Authentication) (err error) {
+	namespace := instance.Namespace
+	reqLogger := log.WithValues("func", "ReconcileRoute", "namespace", namespace)
+	reqLogger.Info("Determined platform-id-auth Route should not exist; removing if present")
+	observedRoute := &routev1.Route{}
+	err = r.Get(ctx, types.NamespacedName{Name: "platform-id-auth", Namespace: namespace}, observedRoute)
+	if errors.IsNotFound(err) {
+		return
```

Comment:
If the error received from attempting to GET the idauth Route is an `IsNotFound`, it still ends up being returned to `reconcileRoute`, which leads to a requeue with error. The error should instead be set to `nil` because failing to get the Route for this reason is exactly what we want - we don't want the Route to be there.

```suggestion
		return nil
```

Reply:
Ah, I see what you mean by requeue now. Yes, since it's not found, we return nil, and no error is reported because that's the expected behaviour.